src/api/java/com/minecolonies/api/colony/workorders/IWorkOrder.java

```diff
@@ -13,12 +15,18 @@
 public interface IWorkOrder
 {
     /**
-     * Read the WorkOrder data from the CompoundNBT.
+     * Get the ID of the work order.
      *
-     * @param compound NBT Tag compound
-     * @param manager  the workManager calling this method.
+     * @return ID of the work order
      */
-    void read(@NotNull CompoundNBT compound, IWorkManager manager);
+    int getID();
+
+    /**
+     * Set the ID of the work order.
+     *
+     * @param id the new ID for the work order
+     */
+    void setID(int id);
```

Comment:
isnt an ID usually immutable? not sure if we should be able to set it

Reply:
The ID isn't actually passed along in the constructor, it's set later and I didn't touch this behaviour.
I can take a second look at it though to see if that can be altered

Reply:
There's 1 usage of this method but it serves 2 purposes, when a new work order is created it's not yet added into the work manager, so we cannot resolve it's ID yet, it's also used when reading back from NBT, in that case you would know it's ID. So for at least 1 case the setter would be required.
Line is [here](https://github.com/ldtteam/minecolonies/blob/bf5df45f6521c6f988ecbce0539fd383634e54af/src/main/java/com/minecolonies/coremod/colony/workorders/WorkManager.java#L272)